src/libs/fsdrv/lv_fs_littlefs.c

```diff
@@ -185,4 +185,76 @@ static lv_fs_res_t fs_tell(lv_fs_drv_t * drv, void * file_p, uint32_t * pos_p)
     return (int32_t)(*pos_p) < 0 ? LV_FS_RES_UNKNOWN : LV_FS_RES_OK;
 }
 
+/**
+ * Open a directory
+ * @param drv       pointer to a driver where this function belongs
+ * @param path      path to the directory beginning with the driver letter (e.g. S:/folder)
+ * @return          a directory descriptor or NULL on error
+ */
+static void * fs_dir_open(lv_fs_drv_t * drv, const char * path)
+{
+    LittleDirectory * ld = (LittleDirectory *)lv_malloc(sizeof(LittleDirectory));
+    LV_ASSERT_NULL(ld);
+
+    lfs_t * lfs = (lfs_t *)drv->user_data;
+    int err = lfs_dir_open(lfs, &ld->dir, path);
+    if(err != LFS_ERR_OK) {
+        return NULL;
+    }
+
+    return (void *)ld;
+}
+
+/**
+ * Close an opened directory
+ * @param drv      pointer to a driver where this function belongs
+ * @param dir_p    pointer to a dir_p variable. (opened with fs_dir_open)
+ * @return         LV_FS_RES_OK: no error or any error from @lv_fs_res_t enum
+ */
+static lv_fs_res_t fs_dir_close(lv_fs_drv_t * drv, void * dir_p)
+{
+    LittleDirectory * ld = (LittleDirectory *)dir_p;
+
+    lfs_t * lfs = (lfs_t *)drv->user_data;
+    int rc = lfs_dir_close(lfs, &ld->dir);
+
+    return rc < 0 ? LV_FS_RES_UNKNOWN : LV_FS_RES_OK;
+}
+
+/**
+ * Read data from an opened directory
+ * @param drv      pointer to a driver where this function belongs
+ * @param dir_p    pointer to a dir_p variable. (opened with fs_dir_open)
+ * @param fn       pointer to a buffer to store the filename
+ * @param fn_len   length of the buffer to store the filename
+ * @return         LV_FS_RES_OK: no error or any error from @lv_fs_res_t enum
+ */
+static lv_fs_res_t fs_dir_read(lv_fs_drv_t * drv, void * dir_p, char * fn, uint32_t fn_len)
+{
+    LittleDirectory * lf = (LittleDirectory *)dir_p;
+
+    lfs_t * lfs = (lfs_t *)drv->user_data;
+
+    int res;
+    struct lfs_info info;
+
+    fn[0] = '\0';
+    do {
+        res = lfs_dir_read(lfs, &lf->dir, &info);
+
+        if(info.name[0] == 0) break; /* End of the directory */
+
+        lv_strncpy(fn, info.name, fn_len);
```

Comment:
Let's use strlcpy. It  makes sure NULL at the end.

```suggestion
        lv_strlcpy(fn, info.name, fn_len);
```

Reply:
I'm not sure null is needed at the end since we are using fn_len :thinking:

Reply:
All the other fs drivers terminate the `fn`. It's kind of an expectation at this point. Do you know the difference between `strncpy` and `strlcpy`? `strncpy` is misleading in what it does, so we're trying to use `strlcpy` to get more reasonable behavior. [See this](https://github.com/lvgl/lvgl/pull/6204).

You'll need to merge/rebase to get `lv_strlcpy`. Or for now you can do what I did in the my suggested change in my review.

Reply:
You're right, forgot to update my fork before starting my PR :(